extensions/machinepools/machinepools.go

```diff
@@ -54,6 +58,14 @@ func MatchNodeRolesToMachinePool(nodeRoles NodeRoles, machinePools []apisV1.RKEM
 		return index, count
 	}
 
+	if nodeRoles.Windows {
+		for index, machinePoolConfig := range machinePools {
```

Comment:
can you just add a comment as to why there's a for loop so we know why there's 2 in the function, otherwise lgtm

Reply:
Added a comment here 👍🏿